Monika After Story/game/script-apologies.rpy

```diff
@@ -0,0 +1,250 @@
+
+    #She knows what you are apologizing for
+    elif mas_apology_reason:
+        m 1eka "Thank you for apologizing for [apology_reason]"
+        m "I accept your apology, [player]. It means a lot to me."
+
+    #She knows that you've got something else to apologize for, and wants you to own up
+    elif persistent._mas_apology_time_db != {}:
+        m 2tfc "[player], if you have something to apologize for, please just say it."
+        m 2rfc "It means a lot more if you'd just admit what you did."
+
+    #She knows there's a reason for your apology but won't comment on it
+    else:
+        #Since this 'reason' technically varies, we don't really have a choice as we therefore can't add 0 to the db
+        #So recover a tiny bit of affection
+        $ mas_gainAffection(modifier=0.1)
+        m 2tkd "What you did wasn't funny, [player]."
+        m 2dkd "Please be more considerate about my feelings in the future."
+
+    #We only want this for actual apology reasons. Not the 0 case or the None case.
+    if mas_apology_reason:
+        #Update the apology_reason count db (if not none)
+        $ persistent._mas_apology_reason_use_db[mas_apology_reason] = persistent._mas_apology_reason_use_db.get(mas_apology_reason,0) + 1
+
+        if persistent._mas_apology_reason_use_db[mas_apology_reason] == 1:
+            #Restore a little bit of affection
+            $ mas_gainAffection(modifier=0.2)
+        elif persistent._mas_apology_reason_use_db[mas_apology_reason] == 2:
+            #Restore a little less affection
+            $ mas_gainAffection(modifier=0.1)
+
+        #Otherwise we recover no affection.
+
+    #Reset the apology reason
+    $ mas_apology_reason = None
+    return
+
+init 5 python:
+    addEvent(
+        Event(
+            persistent.apology_database,
+            eventlabel="mas_apology_bad_nickname",
+            prompt="...for calling you a bad name.",
+            unlocked=False
+        ),
+        code="APL"
+    )
+
+label mas_apology_bad_nickname:
+        if mas_getEV('mas_apology_bad_nickname').shown_count == 0:
+            $ mas_gainAffection(modifier=0.2) # recover a bit of affection
+            m 1eka "Thank you for apologizing for the name you tried to give me."
+            m 2ekd "That really hurt, [player]..."
+            m 2dsc "I accept your apology, but please don't do that again. Okay?"
+            $ mas_unlockEventLabel("monika_affection_nickname")
+
+        elif mas_getEV('mas_apology_bad_nickname').shown_count == 1:
+            $ mas_gainAffection(modifier=0.1) # recover less affection
+            m 2dsc "I can't believe you did that {i}again{/i}."
+            m 2dkd "Even after I gave you a second chance."
+            m 2tkc "I'm disappointed in you, [player]."
+            m 2tfc "Don't ever do that again."
+            $ mas_unlockEventLabel("monika_affection_nickname")
+
+        elif mas_getEV('mas_apology_bad_nickname').shown_count == 2:
```

Comment:
```suggestion
        else:
```
there really is no reason to specify condition here. We should ensure that this event does not get shown over 3 times.